### PR TITLE
Two small strong-mode/warning fixes

### PR DESCRIPTION
**Pull request by matanlurey**
Removing an unused import and fixing a dynamic inference.

### CODE REVIEW

File: lib/src/stream_subscription_transformer.dart [hunk 1]
```diff
@@ -101,5 +100,5 @@ class _TransformedSubscription<T> implements StreamSubscription<T> {
   }
 
   Future/*<E>*/ asFuture/*<E>*/([/*=E*/ futureValue]) =>
-      _inner?.asFuture(futureValue) ?? new Completer().future;
+      _inner?.asFuture(futureValue) ?? new Completer/*<E>*/().future;
```
Comment (nex3) [inline lib/src/stream_subscription_transformer.dart]: @jmesserly Shouldn't this be inferred?

Comment (jmesserly) [inline lib/src/stream_subscription_transformer.dart]: Hmmm. Not sure, I know we have inference for `??`. Do you know what the error message was?

Comment (jmesserly) [inline lib/src/stream_subscription_transformer.dart]: oooooooooh ... NVM, this is bug: https://github.com/dart-lang/sdk/issues/27114

Comment (nex3) [inline lib/src/stream_subscription_transformer.dart]: `Unsafe implicit cast from 'Future<dynamic>' to 'Future<E>'.`

**Pull request merged**